pkg/propertyprovider/azure/provider_integration_test.go

```diff
@@ -132,167 +132,175 @@ var (
-				// calculation is done using the latest pricing data. Inconsistency
-				// should seldom occur though.
-				totalCost := 0.0
-				missingSKUSet := map[string]bool{}
-				isPricingDataStale := false
-
-				pricingDataLastUpdated := pp.LastUpdated()
-				pricingDataBestAfter := time.Now().Add(-trackers.PricingDataShelfLife)
-				if pricingDataLastUpdated.Before(pricingDataBestAfter) {
-					isPricingDataStale = true
-				}
+			if isPricingDataStale {
+				costCollectionWarnings = append(costCollectionWarnings,
+					fmt.Sprintf("the pricing data is stale (last updated at %v); the system might have issues connecting to the Azure Retail Prices API, or the current region is unsupported", pricingDataLastUpdated),
+				)
+			}
 
-				for idx := range nodes {
-					node := nodes[idx]
-					sku := node.Labels[trackers.AKSClusterNodeSKULabelName]
-					cost, found := pp.OnDemandPrice(sku)
-					if !found || cost == pricing.MissingPrice {
-						missingSKUSet[sku] = true
-						continue
-					}
-					totalCost += cost
-				}
-				missingSKUs := []string{}
-				for sku := range missingSKUSet {
-					missingSKUs = append(missingSKUs, sku)
-				}
-				slices.Sort(missingSKUs)
-
-				perCPUCoreCost := "0.0"
-				perGBMemoryCost := "0.0"
-				costCollectionWarnings := []string{}
-				var costCollectionErr error
-
-				switch {
-				case len(nodes) == 0:
-				case totalCost == 0.0:
-					costCollectionErr = fmt.Errorf("nodes are present, but no pricing data is available for any node SKUs (%v)", missingSKUs)
-				case len(missingSKUs) > 0:
-					costCollectionErr = fmt.Errorf("no pricing data is available for one or more of the node SKUs (%v) in the cluster", missingSKUs)
-				default:
-					perCPUCoreCost = fmt.Sprintf(CostPrecisionTemplate, totalCost/totalCPUCores)
-					perGBMemoryCost = fmt.Sprintf(CostPrecisionTemplate, totalCost/totalMemoryGBs)
-				}
+			wantProperties := map[clusterv1beta1.PropertyName]clusterv1beta1.PropertyValue{
+				propertyprovider.NodeCountProperty: {
+					Value: fmt.Sprintf("%d", len(nodes)),
+				},
+			}
 
-				if isPricingDataStale {
-					costCollectionWarnings = append(costCollectionWarnings,
-						fmt.Sprintf("the pricing data is stale (last updated at %v); the system might have issues connecting to the Azure Retail Prices API, or the current region is unsupported", pricingDataLastUpdated),
-					)
+			if costCollectionErr == nil && isCostsEnabled {
+				wantProperties[PerCPUCoreCostProperty] = clusterv1beta1.PropertyValue{
+					Value: perCPUCoreCost,
 				}
+				wantProperties[PerGBMemoryCostProperty] = clusterv1beta1.PropertyValue{
+					Value: perGBMemoryCost,
+				}
+			}
 
-				wantProperties := map[clusterv1beta1.PropertyName]clusterv1beta1.PropertyValue{
-					propertyprovider.NodeCountProperty: {
-						Value: fmt.Sprintf("%d", len(nodes)),
+			var wantConditions []metav1.Condition
+			switch {
+			case !isCostsEnabled:
+				// Cost calculation has been disabled. No need to add a condition.
```

Comment:
Do we want to test when the cost is disabled, we don't collect?

Reply:
Hi Ryan! Unit tests have been added for these scenarios (`TestCollectWithDisabledFeatures` function in `provider_test.go`).